### PR TITLE
[MARVP-36] Integrate and implement wifi animation

### PR DESCRIPTION
**Pull request by krishk95**
Related To : https://konysolutions.atlassian.net/browse/MARVP-36

### CODE REVIEW

File: KonyAccelerate/actions/mobile/AS_Form_e095a506c25445e09b8c7411ea984f9e.json [hunk 1]
```diff
@@ -16,7 +16,7 @@
         "actions": [{
             "id": "ADD_SNIPPET__g27689a162344fc390a575476c98e2d2",
             "type": "ADD_SNIPPET",
-            "codeSnippet": "this.formPreShowAction();",
+            "codeSnippet": "//this.formPreShowAction();\n",
```
Comment (mishradileep) [inline KonyAccelerate/actions/mobile/AS_Form_e095a506c25445e09b8c7411ea984f9e.json:19]: Remove commented code.

File: KonyAccelerate/controllers/mobile/frmWifiInfoController/frmWifiInfoController.js [hunk 2]
```diff
@@ -1,23 +1,78 @@
...
-    formPreShowAction: function() {
-      let accelerateEventData = kony.store.getItem("accelerateEventData");
-      if(!kony.sdk.isNullOrUndefined(accelerateEventData)){
-        let wifiInformation = accelerateEventData.wifiInformation;
-        let wifiUserName = (wifiInformation.hasOwnProperty("wifi_name")) ? wifiInformation.wifi_name : "";
-        let wifiPassword = (wifiInformation.hasOwnProperty("wifi_password")) ? wifiInformation.wifi_password : "";
-        this.view.lblNetworkName.text = wifiUserName;
-        this.view.lblPasswordValue.text = wifiPassword;
-      }
+    formPostShowAction: function() {
```
Comment (mishradileep) [inline KonyAccelerate/controllers/mobile/frmWifiInfoController/frmWifiInfoController.js]: get function name with context. May be ```formWifiInfoPreShowAction```

Comment (mishradileep) [inline KonyAccelerate/controllers/mobile/frmWifiInfoController/frmWifiInfoController.js]: Put a blank line between 2 functions.

File: KonyAccelerate/controllers/mobile/frmWifiInfoController/frmWifiInfoController.js [hunk 3]
```diff
@@ -1,23 +1,78 @@
...
+            this.view.lblNetworkName.text = wifiUserName;
+            this.view.lblPasswordValue.text = wifiPassword;
+          	this.view.flxWifiImageContainer.isVisible = true;
+          	this.animateWifiCard("62%");
+        } else {
+            fetchObjectData(eventConstants.OBJECT_SERVICE_NAME, eventConstants.WIFI_OBJECT_NAME, {}, this.wifiFetchSuccess.bind(this), this.wifiFetchFailure.bind(this));
+        }
+      	
     },
+    /**
```
Comment (mishradileep) [inline KonyAccelerate/controllers/mobile/frmWifiInfoController/frmWifiInfoController.js:26]: Put a blank line between 2 functions

File: KonyAccelerate/controllers/mobile/frmWifiInfoController/frmWifiInfoController.js [hunk 4]
```diff
@@ -1,23 +1,78 @@
...
+     * @function wifiFetchSuccess
+     * @description This function is invoked in the success of wifi object and used to set the data to form
+     * @param successResponse The success response of the wifi object
+     * @private
+     */
+    wifiFetchSuccess: function(successResponse) {
```
Comment (mishradileep) [inline KonyAccelerate/controllers/mobile/frmWifiInfoController/frmWifiInfoController.js]: Can it be ```wifiServiceSuccess```?

File: KonyAccelerate/controllers/mobile/frmWifiInfoController/frmWifiInfoController.js [hunk 5]
```diff
@@ -1,23 +1,78 @@
...
+            kony.print("Unable to fetch Wifi Info");
+        }
+    },
+    /**
+     * @function wifiFetchFailure
+     * @description This function is invoked in the failure of wifi object and used to set the data to form
+     * @param failureResponse The failure response of the wifi object
+     * @private
+     */
+    wifiFetchFailure: function(failureResponse) {
```
Comment (mishradileep) [inline KonyAccelerate/controllers/mobile/frmWifiInfoController/frmWifiInfoController.js]: Can it be ```wifiServiceFail```?

File: KonyAccelerate/controllers/mobile/frmWifiInfoController/frmWifiInfoController.js [hunk 6]
```diff
@@ -1,23 +1,78 @@
...
-      }
+    formPostShowAction: function() {
+        let accelerateEventData = kony.store.getItem("accelerateEventData");
+        if (!kony.sdk.isNullOrUndefined(accelerateEventData)) {
+            let wifiInformation = accelerateEventData.wifiInformation;
+            let wifiUserName = (wifiInformation.hasOwnProperty("wifi_name")) ? wifiInformation.wifi_name : "";
+            let wifiPassword = (wifiInformation.hasOwnProperty("wifi_password")) ? wifiInformation.wifi_password : "";
+            this.view.lblNetworkName.text = wifiUserName;
+            this.view.lblPasswordValue.text = wifiPassword;
+          	this.view.flxWifiImageContainer.isVisible = true;
```
Comment (mishradileep) [inline KonyAccelerate/controllers/mobile/frmWifiInfoController/frmWifiInfoController.js:18]: I would like to see fade in animation for wifi image. But we can proceed for now. Keep this noted

File: KonyAccelerate/controllers/mobile/frmWifiInfoController/frmWifiInfoController.js [hunk 7]
```diff
@@ -1,23 +1,78 @@
...
+        if (wifiInfo !== null) {
+            let wifiData = wifiInfo[0];
+            let wifiUserName = (wifiData.hasOwnProperty("wifi_name")) ? wifiData.wifi_name : "";
+            let wifiPassword = (wifiData.hasOwnProperty("wifi_password")) ? wifiData.wifi_password : "";
+            this.view.lblNetworkName.text = wifiUserName;
+            this.view.lblPasswordValue.text = wifiPassword;
+          	this.view.flxWifiImageContainer.isVisible = true;
+          	this.animateWifiCard("62%");
+        } else {
+            kony.print("Unable to fetch Wifi Info");
```
Comment (mishradileep) [inline KonyAccelerate/controllers/mobile/frmWifiInfoController/frmWifiInfoController.js]: ```Something went wrong! Please try sometime later. Make sure you have network connectivity```

**Review by mishradileep — CHANGES_REQUESTED** [review]

File: KonyAccelerate/controllers/mobile/frmWifiInfoController/frmWifiInfoController.js (hunk 6 above)
Comment (krishk95) [reply KonyAccelerate/controllers/mobile/frmWifiInfoController/frmWifiInfoController.js:18]: Sure, we will keep this for the last

File: KonyAccelerate/controllers/mobile/frmWifiInfoController/frmWifiInfoController.js (hunk 7 above)
Comment (krishk95) [reply KonyAccelerate/controllers/mobile/frmWifiInfoController/frmWifiInfoController.js]: Fixed

File: KonyAccelerate/controllers/mobile/frmWifiInfoController/frmWifiInfoController.js (hunk 4 above)
Comment (krishk95) [reply KonyAccelerate/controllers/mobile/frmWifiInfoController/frmWifiInfoController.js]: Actually we are using the same naming convention for object service.
"[ObjectName][Operation][Success/Failure]"

File: KonyAccelerate/controllers/mobile/frmWifiInfoController/frmWifiInfoController.js (hunk 5 above)
Comment (krishk95) [reply KonyAccelerate/controllers/mobile/frmWifiInfoController/frmWifiInfoController.js]: Actually we are using the same naming convention for object service.
"[ObjectName][Operation][Success/Failure]"

File: KonyAccelerate/controllers/mobile/frmWifiInfoController/frmWifiInfoController.js (hunk 2 above)
Comment (krishk95) [reply KonyAccelerate/controllers/mobile/frmWifiInfoController/frmWifiInfoController.js]: Addressed

**Review by mishradileep — APPROVED** [review]

File: KonyAccelerate/actions/mobile/AS_Form_e095a506c25445e09b8c7411ea984f9e.json (hunk 1 above)
Comment (vigneshwarankarunanithi) [inline KonyAccelerate/actions/mobile/AS_Form_e095a506c25445e09b8c7411ea984f9e.json:19]: remove this commented code.. if you don't need this action remove the action itself

File: KonyAccelerate/actions/mobile/appEvents.js [hunk 8]
```diff
@@ -1,6 +1,5 @@
 define({
     AS_AppEvents_c081992218f44bf89cef587f76549c96: function AS_AppEvents_c081992218f44bf89cef587f76549c96(eventobject) {
         var self = this;
-        appPreShow();
```
Comment (vigneshwarankarunanithi) [inline KonyAccelerate/actions/mobile/appEvents.js:4]: is the action removed? or only deleted the line. please check and confirm..

Comment (krishk95) [reply KonyAccelerate/actions/mobile/appEvents.js:4]: This action is deleted

File: KonyAccelerate/actions/mobile/AS_Form_e095a506c25445e09b8c7411ea984f9e.json (hunk 1 above)
Comment (krishk95) [reply KonyAccelerate/actions/mobile/AS_Form_e095a506c25445e09b8c7411ea984f9e.json:19]: This is already removed

**Review by vigneshwarankarunanithi — APPROVED** [review]

**Pull request merged**